sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/UnsafeRowParquetRecordReader.java

```diff
@@ -645,7 +676,15 @@ private void decodeDictionaryIds(int rowId, int num, ColumnVector column) {
             }
           } else if (column.dataType() == DataTypes.ByteType) {
             for (int i = rowId; i < rowId + num; ++i) {
-              column.putByte(i, (byte)dictionary.decodeToInt(dictionaryIds.getInt(i)));
+              column.putByte(i, (byte) dictionary.decodeToInt(dictionaryIds.getInt(i)));
+            }
+          } else if (column.dataType() == DataTypes.ShortType) {
+            for (int i = rowId; i < rowId + num; ++i) {
+              column.putShort(i, (short) dictionary.decodeToInt(dictionaryIds.getInt(i)));
+            }
+          } else if (DecimalType.is64BitDecimalType(column.dataType())) {
```

Comment:
This code is like a nested loop. On the outer, we loop over the parquet type and on the inner, the column vector type.

In the case you mentioned, the parquet type is int64 and the column vector type is long so I think this just works. Let me make that more explicit by adding a check in the INT64 branch.